### PR TITLE
[ephemeris] Add support for overriding holiday definitions

### PR DESCRIPTION
**Pull request by jlaur**
The version of Jollyday we are using was abandoned years ago, so the holiday definitions we are using are getting outdated as well.

This pull request introduces the possibility to override holiday definitions in core, and as an example contains updated Danish definitions, ending Great Prayer Day this year, since our government has [abolished](https://www.thelocal.dk/20230228/great-prayer-day-abolished-by-danish-parliament) it from next year.

The obvious disadvantage is that we would then take over the maintenance. But on the other hand, they are not maintained upstream anymore, so at least it would improve the user experience. Additionally, it would not change the API, so upgrading to **focus-shift** later will be seamless for users.

Related to #3544

### CODE REVIEW

File: bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java [hunk 1]
```diff
@@ -231,9 +234,15 @@ private URL getUrl(String filename) throws FileNotFoundException {
     private HolidayManager getHolidayManager(Object managerKey) {
         HolidayManager holidayManager = holidayManagers.get(managerKey);
         if (holidayManager == null) {
-            final ManagerParameter parameters = managerKey.getClass() == String.class
-                    ? ManagerParameters.create((String) managerKey)
-                    : ManagerParameters.create((URL) managerKey);
+            final ManagerParameter parameters;
+            if (managerKey.getClass() == String.class) {
```
Comment (clinique) [inline bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java:238]: Wouldn't `if (managerKey instanceof String stringKey) {` be better ? This avoids below typecasts

**Review by clinique — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java (hunk 1 above)
Comment (jlaur) [reply bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java]: Right - fixed. Do you by any chance know how to run the integration tests locally? I would like to refactor a bit more, but first would need to get the tests running.

**Review by clinique — APPROVED** [review]

**Review by clinique — APPROVED** [review]

File: bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java [hunk 2]
```diff
@@ -107,7 +111,6 @@ public class EphemerisManagerImpl implements EphemerisManager, ConfigOptionProvi
     public EphemerisManagerImpl(final @Reference LocaleProvider localeProvider) {
         this.localeProvider = localeProvider;
 
-        final Bundle bundle = FrameworkUtil.getBundle(getClass());
```
Comment (J-N-K) [inline bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java:110]: I would suggest to add `BundleContext bc` to the constructor parameters and use `Bundle bundle = bundleContext.getBundle();` here. I don't know why it was done in this strange way.

Comment (jlaur) [reply bundles/org.openhab.core.ephemeris/src/main/java/org/openhab/core/ephemeris/internal/EphemerisManagerImpl.java:110]: Done.

**Review by J-N-K — APPROVED** [review]
LGTM

**Pull request merged**